### PR TITLE
Add VPCEndpoints on created VPC

### PR DESCRIPTION
**Pull request by peternied**
### Description
Add VPCEndpoints for the created VPC flow

### Issues Resolved
- Support IPv6 within MA deployed VPC https://opensearch.atlassian.net/browse/MIGRATIONS-2145

### Testing
Deployed to AWS account and logged onto bootstrap box

### Check List
- [X] New functionality includes testing
  - [X] All tests pass, including unit test, integration test and doctest
- [ ] ~New functionality has been documented~
- [X] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: deployment/migration-assistant-solution/package.json [hunk 1]
```diff
@@ -22,8 +22,8 @@
     "@types/eslint__js": "^8.42.3",
     "@types/jest": "^29.5.5",
     "@types/node": "^20.9.0",
-    "aws-cdk": "2.105.0",
-    "aws-cdk-lib": "2.105.0",
+    "aws-cdk": "2.167.2",
```
Comment (peternied) [inline deployment/migration-assistant-solution/package.json:25]: `ipProtocal` was added in a newer version of the CDK, also did the upgrade for it.

File: deployment/migration-assistant-solution/lib/solutions-stack.ts [hunk 2]
```diff
@@ -162,9 +176,33 @@ export class SolutionsInfrastructureStack extends Stack {
...
+            vpc = new Vpc(this, 'Vpc', {
+                ipProtocol: IpProtocol.DUAL_STACK
+            });
+            // S3 used for storage and retrieval of snapshot data for backfills
+            new GatewayVpcEndpoint(this, 'S3VpcEndpoint', {
+                service: GatewayVpcEndpointAwsService.S3,
+                vpc: vpc,
+            });
+
+            const serviceEndpoints = [
```
Comment (peternied) [inline deployment/migration-assistant-solution/lib/solutions-stack.ts:190]: Only added endpoints with substantive amount of traffic.  We had many more in the MA network-stack, but these are really the only ones that would see a price difference.  Its a 1/3 the cost to use the VPCe compared to the NAT Gateway.

File: deployment/cdk/opensearch-service-migration/lib/migration-assistance-stack.ts [hunk 3]
```diff
@@ -180,6 +181,7 @@ export class MigrationAssistanceStack extends Stack {
         const sharedLogsSG = new SecurityGroup(this, 'sharedLogsSG', {
             vpc: props.vpc,
             allowAllOutbound: false,
+            allowAllIpv6Outbound: false,
```
Comment (lewijacn) [inline deployment/cdk/opensearch-service-migration/lib/migration-assistance-stack.ts:184]: Have we verified Ipv6 gets used and does not break the solution?

**Review by AndreKurait — APPROVED** [review]
LGTM, i'd like to also get someone else to weigh in on the security implication of only adding a subset of the VPCe's of aws services we call

File: deployment/cdk/opensearch-service-migration/lib/migration-assistance-stack.ts (hunk 3 above)
Comment (peternied) [reply deployment/cdk/opensearch-service-migration/lib/migration-assistance-stack.ts]: It does seem to break the solution, or at least things are not working correctly with this PR, I've reverted all IPv6 related changes.

**Comment by peternied** [comment]
Just deployed and verified the latest version including the AMI locking changes.

File: deployment/cdk/opensearch-service-migration/lib/migration-assistance-stack.ts (hunk 3 above)
Comment (peternied) [reply deployment/cdk/opensearch-service-migration/lib/migration-assistance-stack.ts:184]: Fixed that issue, fresh deployment of this PR looks good
- https://github.com/opensearch-project/opensearch-migrations/pull/1153

**Pull request merged**